File: classes/proposal.py
from datetime import date, time

class Proposal():
    def __init__(self, id: int, owner_email: str, build_time: int, prefered_dates_start: list[date], prefered_dates_end: list[date], avoid_dates_start: list[date], avoid_dates_end: list[date], night_obs: bool, avoid_sunrise_sunset: bool, minimum_antennas: int, lst_start_time: time, lst_start_end_time: time, simulated_duration: int, score: float) -> None:
        self.id: int = id
        self.owner_email: str = owner_email
        self.build_time: int = build_time
        self.prefered_dates_start: list[date] = prefered_dates_start
        self.prefered_dates_end: list[date] = prefered_dates_end
        self.avoid_dates_start: list[date] = avoid_dates_start
        self.avoid_dates_end: list[date] = avoid_dates_end
        self.night_obs: bool = night_obs
        self.avoid_sunrise_sunset: bool = avoid_sunrise_sunset
        self.minimum_antennas: int = minimum_antennas
        self.lst_start_time: time = lst_start_time
        self.lst_start_end_time: time = lst_start_end_time
        self.simulated_duration: int = simulated_duration
        self.score: int = score

    @staticmethod
    def parse_time(time_str: str) -> time:
        """
        Parse a time string in the format "HH:MM" and return a datetime.time object.
        """
        hour, minute = map(int, time_str.split(":"))
        return time(hour, minute)
    
    def get_score(self):
        return 1

File: frontend/src/app/interfaces/proposal.ts

export interface Proposal {
    id: number;
    description: string;
    proposal_id: string;
    owner_email: string;
    instrument_product: string;
    instrument_integration_time: number;
    instrument_band: string;
    instrument_pool_resources: string;
    lst_start_time: string;
    lst_start_end_time: string;
    simulated_duration: number;
    night_obs: boolean;
    avoid_sunrise_sunset: boolean;
    minimum_antennas: number;
    general_comments: string | null;
    prefered_dates_start_date: Date[];
    prefered_dates_end_date: Date[];
    avoid_dates_start_date: Date[];
    avoid_dates_end_date: Date[];
    score: number;
    scheduled_start_datetime: Date | null;
}
